webv2/web.go

```diff
@@ -988,6 +1036,67 @@ func dropRule(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func setShardIdColumnAsPrimaryKey(isAddedAtFirst bool) {
+	sessionState := session.GetSessionState()
+	if isAddedAtFirst {
+		for _, table := range sessionState.Conv.SpSchema {
+			pkRequest := primarykey.PrimaryKeyRequest{
+				TableId: table.Id,
+				Columns: []ddl.IndexKey{{ColId: table.ShardIdColumn, Order: 1}},
+			}
+			for index := range table.PrimaryKeys {
+				pk := table.PrimaryKeys[index]
+				pkRequest.Columns = append(pkRequest.Columns, ddl.IndexKey{ColId: pk.ColId, Order: pk.Order + 1, Desc: pk.Desc})
+			}
+			primarykey.UpdatePrimaryKeyAndSessionFile(pkRequest)
+		}
+	} else {
+		for _, table := range sessionState.Conv.SpSchema {
+			size := len(table.PrimaryKeys)
+			pkRequest := primarykey.PrimaryKeyRequest{
+				TableId: table.Id,
+				Columns: table.PrimaryKeys,
+			}
+			pkRequest.Columns = append(pkRequest.Columns, ddl.IndexKey{ColId: table.ShardIdColumn, Order: size + 1})
```

Comment:
seems like a bunch of code in the 2 blocks around iterating over tables etc is common. Can we use common code and move the check inside the loop ?